### DISCUSSION TITLE
Shape functions in FV Elemental Kernel

### DISCUSSION THREAD

**Discussion by heinono1** [idaholab/moose#25248] [Q&A General] [answered]
Hi,

How can I access shape functions _phi[_j] and _grad_phi[_j] in an FV elemental kernel?

**Comment by GiudGiud** [comment, marked as answer]
Hello

I dont think you can, they're both unnecessary at this point in time where we only do constant monomial FV variable. 
grad is 0, phi is constant. 
what are you planning to do?

Guillaume

**Reply by heinono1** [reply]
Oh right. I was thinking of an approximation to the functional derivative of an energy term that has the square of the gradient. That is a term that leads to a flux kernel in FV and if I could replace the functional derivative with terms that only has first-order derivatives and lead to Elemental terms, I could see what the difference in timing is. There is a "Box" approximation for that functional derivative (used in the old Magpar code) but the price is a gradient of the shape function. But I'll drop that thought.

**Reply by GiudGiud** [reply]
See my comment on the other thread. 
Are you coding the weak form in FV?
We can do variable.gradient(elem_arg) to get an elemental gradient btw

**Reply by heinono1** [reply]
I am doing strong form of FV. Yes I've seen (and used) the elemental gradient form. I have to think a little bit more about how to use the "Box" approximation sensibly.
